### DISCUSSION TITLE
Libre2 sensor-no BT connection with xdrip lately

### DISCUSSION THREAD

**Discussion by marinko1973** [NightscoutFoundation/xDrip#3407] [Bug reports] [answered]
For some unknown reason xdrip is not picking up Bluetooth connection from libre 2 sensor. I am able to complete NFC scan and like that get the readings on the xdrip but after that no further BT readings are forthcoming. have restarted mobile for a number of time and reinstalled xdrip and oo2 app to latest version. I even tried rolling back to some older 2024 version , however that did not help. this behavior is on my daughter's phone Xiaomi 13t with latest Android . however if I use my own phone Samsung a54 everything is in order. have tried to compare logs between two phones and the difference what I can see is that LibreoopAlgorithm line is not listed in daughter's phone log. It is worth mentioning that the xdrip and oo2 was running on daughter's phone very fine until we replaced last sensor.

**Comment by marinko1973** [comment]
![Screenshot_2024-03-28-21-35-48-815_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/f9d9978d-4bb2-499b-b263-365ac60c8b4d)
![Screenshot_2024-03-28-21-35-41-979_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/1d686922-ac52-4b12-93b2-1f4005f59376)
![Screenshot_2024-03-28-21-35-30-591_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/cc4782e3-1c0b-4967-9c37-79bc9c64c48e)
![Screenshot_2024-03-28-21-35-26-730_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/58a7717e-21f2-40f9-ae24-bc95d172c360)
![Screenshot_2024-03-28-21-35-10-406_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/5644d5f5-0258-4c31-9ca2-60463e889bd8)

**Comment by marinko1973** [comment]
![Screenshot_2024-03-28-21-52-02-385_com hg4 oopalgorithm oopalgorithm2](https://github.com/NightscoutFoundation/xDrip/assets/68336031/6072b5c3-059e-4f39-997c-95442dec9e28)
![Screenshot_2024-03-28-21-51-45-607_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/895e6ad4-d058-4cdd-9a67-249d3834935e)
![Screenshot_2024-03-28-21-51-39-076_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/68336031/d91a5caa-4f9f-423f-940f-c8f2ece433fd)

**Reply by PhilTU** [reply]
The "direct intent" option in OOP2 is for WearOS I think, using Foreground mode can avoid it being shut down.

What region is sensor from?

I note the 21:28 error fail to set up Bluetooth streaming

**Reply by marinko1973** [reply]
EU region

**Reply by marinko1973** [reply]
I Will try with foreground mode, though that was not the setup at the time evrything was in order before the last sensor renplacement.

**Reply by PhilTU** [reply]
Yeah the Foreground is a precaution against being shut down.

All the screen shots are from the faulty phone? And the same sensor works with xDrip on a different phone?

Will look later on a bigger screen.

**Reply by PhilTU** [reply]
This is the failure mode I noticed. Can you say what started the sensor and has it been neutralised ie if you previously used other phone on this sensor if it is present turn off the Bluetooth and try NFC scan again. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/76240903/fc552bef-c3d7-4c17-92dc-33808a0907bc)

**Reply by marinko1973** [reply]
Screen shootw are from the faulty phone and the same sensor works with xdrip on a different phone. Sensor was started with the libre one reader as always. My phone is the other phone i used and is always around. I Will try turning of Bluetooth on my phone and sve if that will help.

**Reply by PhilTU** [reply]
Yes, you could set xDrip on the "working phone" to Hardware Data Source of "disable collection" to prevent interference.

**Reply by marinko1973** [reply]
I have followed all advices and have come to understand that my daughters phone is picking up Bluetooth data from libre 2 sensor only if it is been kept in close proximity of the libre 2 sensor. The Bluetooth data connection from sensor is lost as soon as her phone is moved 3 feet away. I am clueless about this behaviour and could only think that that her phone could have a hardware issue. This is on the other hand unlikly since other BT devices have no issues communicating with the same phone.

**Reply by PhilTU** [reply]
Interesting. The Libre 2 has a low output power and BLE range may be less than stronger signals. Libre 3 has more power. 

Keeping the phone on the same side as the sensor. No case. Minimal obstruction between sensor and phone. Minimise other Bluetooth devices and 2.4 GHz WiFi. 

What is the phone? Samsung had a bug last year in A53 which reduced Bluetooth range to cm so it may be worth looking at that to see if a software update (or selection of an older driver) may be helpful. 

I also put the sensor on so the antenna is at the bottom below the battery.

**Reply by marinko1973** [reply]
It is xiaomi 13t. But what i dont uderstand is that the same phone was reading libre 2 sensors just fine, even from the other room, until the last sensor was put in use. I will have to go through the log again and look for the clue.

**Reply by marinko1973** [reply]
I will check when was the last update made on her phone

**Reply by PhilTU** [reply]
It is odd, but things fail. May just be a defective sensor.

One of the BLE scanner apps would give you a received signal strength reading you could compare phones to see if there's an issue with the Bluetooth system - compare received signal strength from a speaker or something similar.

**Reply by marinko1973** [reply, marked as answer]
I give up on searching the reason for fault in BT communication between libre 2 sensor and xiaomi 13t mob phone. After using BTE scanner on two phones (xiaomi 13t/non working and Samsung a54/working) it appers that xiaomi is significantly inferior in receiving any BT signal when compered with Samsung a54. My phone samsung a54 has never had any issue connecting to libre 2 sensor while at the same time xiamo could maintain some communication only if it is been kept in very close proximity of the sensor. I am now reconsidering to replace xiaomi/daughter's phone with a Samsung. Thanks PhilTU for all the help provided.

**Comment by Navid200** [comment]
@PhilTU
